src/files/files.ts

```diff
@@ -0,0 +1,250 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+import { convertToExcalidrawElements } from '@excalidraw/excalidraw'
+import type {
+	BinaryFileData,
+	DataURL,
+	ExcalidrawImperativeAPI,
+} from '@excalidraw/excalidraw/types/types'
+import { Collab } from '../collaboration/collab'
+import type { FileId } from '@excalidraw/excalidraw/types/element/types'
+
+type Meta = {
+	name: string,
+	type: string,
+	lastModified: number,
+	dataurl: string,
+	fileId: string,
+}
+
+export class FileHandle {
+
+	private collab: Collab
+	private excalidrawApi: ExcalidrawImperativeAPI
+	private types: string[]
+	constructor(
+		excalidrawApi: ExcalidrawImperativeAPI,
+		collab: Collab,
+		types: string[],
+	) {
+		this.collab = collab
+		this.excalidrawApi = excalidrawApi
+		this.types = types
+		const containerRef = document.getElementsByClassName(
+			'excalidraw-container',
+		)[0]
+		const constructedFile: BinaryFileData = {
```

Comment:
I'm not sure I fully follow how/why this works with the placeholder image, maybe you can do a short explanation of how the file is stored to the PR description also for future reference.